### ISSUE TITLE
Resolve labels overlapping in BarGauge and Funnel

### ISSUE THREAD

**Issue by pavelgruba** [DevExpress/DevExtreme#7167]
# The Problem

Labels in the Funnel and BarGauge widgets can overlap each other but the widgets do not provide any options to avoid this.  
![labels_issue](https://user-images.githubusercontent.com/19471748/53814275-62eb8500-3f70-11e9-8701-c441cc161526.png)

# The Proposed Solution

We will introduce a new option, `resolveLabelOverlapping`, that specifies how the widget should behave when labels overlap each other.

In the Funnel, this option accepts `shift` (default), `hide`, and `none` as values. If there is still no room to display all the labels after they are shifted, the algorithm hides certain labels.

In the BarGauge, `resolveLabelOverlapping` will accept `hide` (default) and `none`.

The following images illustrate the different modes:

```js
let funnelOptions = {
    resolveLabelOverlapping: "shift"
};
```
![funnel_shift](https://user-images.githubusercontent.com/19471748/53814788-62072300-3f71-11e9-96f8-899fd2b91491.png)

```js
let funnelOptions = {
    resolveLabelOverlapping: "hide"
};
```
![funnel_hide](https://user-images.githubusercontent.com/19471748/53814868-8cf17700-3f71-11e9-8546-2044d24e0e00.png)

```js
let barGaugeOptions = {
    resolveLabelOverlapping: "hide"
};
```
![image](https://user-images.githubusercontent.com/19471748/53815014-dcd03e00-3f71-11e9-857d-f068bc7a128f.png)

# We Need Your Feedback

## Take a Quick Poll

[Do you find the `resolveLabelOverlapping` option helpful?](https://docs.google.com/forms/d/e/1FAIpQLScZVhKJre6xD8_iP5ZA5A2SAGzdG_xDhAqyDES88pkY57zWBg/viewform?usp=sf_link)

## Live Sandbox

Funnel examples - [jQuery](https://codepen.io/pavelgruba/pen/bZYyLr?editors=0010), [Angular](https://codesandbox.io/s/l2qry584rz)

# Get Notified of Updates

Subscribe to this thread or to our [Facebook](https://www.facebook.com/DevExpress.DevExtreme/) and [Twitter](https://twitter.com/devextreme) accounts for updates on this topic.

**Comment by pavelgruba** [comment]
**Update:** The Live Sandbox section has been added.

**Comment by pavelgruba** [comment]
These features are now available in the [v19.1](https://github.com/DevExpress/DevExtreme/releases/tag/19.1.3) release. If you have questions or find any bugs, feel free to create new GitHub [issues](https://github.com/DevExpress/DevExtreme/issues) or tickets in our [Support Center](https://www.devexpress.com/Support/Center/).
